services/travis/travis-build.service.js

```diff
@@ -1,26 +1,25 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
-const { checkErrorResponse } = require('../../lib/error-helper')
-const log = require('../../lib/log')
+const Joi = require('joi')
 
-// Handle .org and .com.
-//
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class TravisBuild extends LegacyService {
+const BaseSvgScrapingService = require('../base-svg-scraping')
+
+const schema = Joi.object({
+  message: Joi.string()
+    .regex(/^[a-z]+$/)
```

Comment:
is there a smaller set of values we could test against? the tester is using the `isBuildStatus` from `test-validators.js`

Reply:
I mentioned it as I remembered seeing a similar pattern for the Azure DevOps builds (which is a badge scraping implementation) https://github.com/badges/shields/blob/master/services/azure-devops/azure-devops-helpers.js#L9 where there is validation.

I suppose it's a little philosophical, but IMO I'm not sure why we'd have prod validation for some services but not others. I think we're following the same pattern in other (JSON API based) services where the returned value is the source of truth, but we still have schema validation. 

For example if Travis were to change their status message from `passing` to `passed` tomorrow, what would happen to the badges in prod (in particular the color scheme)? Would the daily test run catch this to let us know?

I don't feel strongly about it either way, but if we are going to establish a practice of not doing validation in certain scenarios (for example badge scraping services) then we should document that somewhere.

Reply:
To be honest I was not aware that we had cases where we do perform very strict validation in production, generally we don't seem to do that with build badges (Wercker, CircleCi, AppVeyor, BitbucketPipelines to name just a few).

If Travis were to change their status message from `passing` to `passed`, I would much rather have users reporting that the colour is wrong rather than having to do emergency fixing because the badge is completely broken and displays `inaccessible` or something like that. ^^

But you're right that we should try to be consistent, let's wait for someone else's opinion. 😉

Reply:
I'm a fan of failing loudly because it means issues are more likely to be reported and fixed. I'm also a fan of adapting to specific circumstances, like starting with something strict and loosening it if need be. If we find we're deploying emergency patches for schema validations more than once, then we could switch tacks, leaving it loose, and basically hoping the colors are right.

Reply:
That sounds like a good plan. I did not realize we were handling it so inconsistently.

Reply:
I'll raise an issue once we're happy with this as it will help narrow down the work that has to be done.

I've provided stronger validation and tried to extract things as much as possible so that can easily be reused in different places. Let me know what you think!

Reply:
> I would much rather have users reporting that the colour is wrong rather than having to do emergency fixing because the badge is completely broken and displays inaccessible or something like that

That's a fair point too. Ultimately I suppose it just boils down to the tradeoffs between more/less strict validation. I'm good with whatever we decide 👍 but agree consistency should be our target. 

If I'm following correctly it sounds like we want to lean towards more strict but also reasonably support exceptions, for example if the validation becomes to brittle for a particular service?